src/main/kotlin/inc/evil/d17map/web/filters/GlobalExceptionHandlingFilter.kt

```diff
@@ -0,0 +1,62 @@
+package inc.evil.d17map.web.filters
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import inc.evil.d17map.exceptions.ErrorCodes
+import inc.evil.d17map.exceptions.ErrorResponse
+import inc.evil.d17map.exceptions.JWTFilterException
+import io.github.oshai.kotlinlogging.KotlinLogging
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+
+
+private val kotlinLogger = KotlinLogging.logger {}
+
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@Component
+class GlobalExceptionHandlingFilter(private val objectMapper: ObjectMapper) : OncePerRequestFilter() {
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        try {
+            filterChain.doFilter(request, response)
+        } catch (ex: Exception) {
+            handleException(response, ex)
+        }
+    }
+
+    private fun handleException(
+        response: HttpServletResponse,
+        ex: Exception,
+    ) {
+        kotlinLogger.error(ex) { "Exception caught in GlobalExceptionHandlingFilter: ${ex.message}" }
+
+        val errorResponse = when (ex) {
+            is JWTFilterException -> {
+                response.status = HttpServletResponse.SC_UNAUTHORIZED
+                ErrorResponse(
+                    errorCode = ErrorCodes.UNAUTHORIZED_ERROR,
+                    message = "${ex.message}"
+                )
+            }
+            else -> {
+                response.status = HttpServletResponse.SC_INTERNAL_SERVER_ERROR
+                ErrorResponse(
+                    errorCode = ErrorCodes.UNKNOWN_ERROR,
+                    message = "Internal Server Error: ${ex.message}"
+                )
+            }
```

Comment:
Czy nie padną tutaj inne wyjątki? IllegalArgumentException, AccessDeniedException?

Reply:
Mamy 3 punkty obsługi błędów.
1. @ControllerAdvice - złapie wszystko co się stanie już po filtrach

2. ExceptionTranslationFilter - łapie błędy autentykacji I autoryzacji.

3. Ten filtr tutaj - łapie wszystko inne, tzn u nas błędy JWT, bo Ten filtr jest przed 2.

Gdybyśmy łapali wszystko tylko tutaj, to jak najbardziej miałabyś rację, ale tak to nie widzę sensu łapania tych co są już wcześniej złapane.

Dodam też, że logger daje nam wygląd w StackTrace, bo np błąd Serializacji jest błędem typu Internal Server Erorr z punktu widzenia frontu. Więc my I tak zobaczymy co rzuciło Ten wyjątek.